suite-native/module-authorize-device/src/components/connect/PinFormControlButtons.tsx

```diff
@@ -43,25 +29,28 @@ const buttonsWrapperStyle = prepareNativeStyle(utils => ({
 
 const ANIMATION_DURATION = 100;
 
-export const PinFormControlButtons = () => {
+export type PinFormControlButtonsProps = {
+    onSuccess?: () => void;
+};
+
+export const PinFormControlButtons = ({ onSuccess }: PinFormControlButtonsProps) => {
     const [containerHeight, setContainerHeight] = useState(0);
     const animatedHeight = useSharedValue(0);
 
     const dispatch = useDispatch();
     const { applyStyle } = useNativeStyles();
 
     const openLink = useOpenLink();
-    const navigation = useNavigation<NavigationProp>();
     const { showAlert } = useAlert();
     const { handleSubmit, getValues, watch, setValue, reset } = useFormContext();
     const hasDeviceAuthFailed = useSelector(selectDeviceAuthFailed);
 
     const handleSuccess = useCallback(() => {
-        if (navigation.canGoBack()) {
-            navigation.goBack();
+        if (onSuccess) {
+            onSuccess();
```

Comment:
you can replace the whole if with:
 ```suggestion
            onSuccess?.();
```

Reply:
Nice. Now I see we use it on many places. 👍